showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITEndpointContext.java

```diff
@@ -50,17 +130,72 @@ public void endpointResolution_default() throws InterruptedException, IOExceptio
 
   // User configuration
   @Test
-  public void endpointResolution_userConfiguration() throws InterruptedException, IOException {
+  public void endpointResolution_userSetEndpoint() throws InterruptedException, IOException {
     String customEndpoint = "test.com:123";
     EchoClient echoClient = null;
     try {
       EchoSettings echoSettings =
-          EchoSettings.newBuilder()
+          ExtendedEchoSettings.newBuilder()
+              .setCredentialsProvider(NoCredentialsProvider.create())
+              .setEndpoint(customEndpoint)
+              .build();
+      echoClient = EchoClient.create(echoSettings);
+      Truth.assertThat(echoClient.getSettings().getEndpoint()).isEqualTo(customEndpoint);
+      Truth.assertThat(echoClient.getSettings().getUniverseDomain())
+          .isEqualTo(GoogleCredentials.GOOGLE_DEFAULT_UNIVERSE);
+    } finally {
+      if (echoClient != null) {
+        echoClient.close();
+        echoClient.awaitTermination(
+            TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+      }
+    }
+  }
+
+  @Test
+  public void endpointResolution_userSetUniverseDomain() throws IOException, InterruptedException {
+    String customUniverseDomain = "random.com";
+    EchoClient echoClient = null;
+    try {
+      EchoSettings echoSettings =
+          ExtendedEchoSettings.newBuilder()
+              .setCredentialsProvider(NoCredentialsProvider.create())
+              .setUniverseDomain(customUniverseDomain)
+              .build();
+      echoClient = EchoClient.create(echoSettings);
+      // If user configured the universe domain, the endpoint is constructed from it
+      Truth.assertThat(echoClient.getSettings().getEndpoint()).isEqualTo("test.random.com:443");
+      Truth.assertThat(echoClient.getSettings().getUniverseDomain())
+          .isEqualTo(customUniverseDomain);
+    } finally {
+      if (echoClient != null) {
```

Comment:
Extract `echoClient` to a field and put these logic in the clean up method?

Reply:
Ah good idea. Will do!